weave/integrations/openai/openai_test.py

```diff
@@ -0,0 +1,773 @@
+import os
+import typing
+
+import pytest
+from openai import AsyncOpenAI, OpenAI
+
+import weave
+from weave.trace_server import trace_server_interface as tsi
+
+from .openai_sdk import openai_patcher
+
+model = "gpt-4o"
+
+
+def filter_body(r: typing.Any) -> typing.Any:
+    r.body = ""
+    return r
+
+
+@pytest.mark.skip_clickhouse_client  # TODO:VCR recording does not seem to allow us to make requests to the clickhouse db in non-recording mode
+@pytest.mark.vcr(
+    filter_headers=["authorization"],
+    allowed_hosts=["api.wandb.ai", "localhost"],
+    before_record_request=filter_body,
+)
+def test_openai_quickstart(client: weave.weave_client.WeaveClient) -> None:
```

Comment:
for all tests, can you please add assertions on: `op_name` and `finished_at` != None and `finished_at` < `started_at`

Reply:
will do. didn't do it originally because the op name were not setting to the desired name. now that it's happening, adding these checks too.

Reply:
Can you please add these assertions (: